beancount.el

```diff
@@ -34,6 +34,7 @@
 (require 'subr-x)
 (require 'outline)
 (require 'thingatpt)
+(require 'imenu)
```

Comment:
`imenu-generic-expression` is defined in `imenu.el` and used in `beancount.el`, so my assumption that the require was needed in `beancount.el` in addition to (implicitly) `beancount-tests.el`.

If this is not the case, then I'll move it to `beancount-tests.el`.

Reply:
`imenu-generic-expression` is declared as a local variable in `imenu.el`, but you are updating the variable using `setq` here. If `imenu.el` has not been loaded yet, it will set the global value.

If you don't require `imenu`, I think you should set it locally, e.g. using `setq-local`...

FYI: In most cases, you would declare a variable from an external package using `defvar` without an initial value:

```emacs-lisp
(defvar imenu-generic-expression)
```

See https://github.com/purcell/package-lint/issues/205 for details.